rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -27,168 +57,154 @@ func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset
 
 func Qualify_AssetClass_ForeignExchange: <"Qualifies a product as having the Asset Class classification Foreign Exchange">
     inputs:
         economicTerms EconomicTerms (1..1)
     output:
         is_product boolean (1..1)
 
-    alias optionUnderlier: economicTerms -> payout -> optionPayout only-element -> underlier
-    alias forwardUnderlier:
-        economicTerms -> payout -> forwardPayout only-element -> underlier
+    alias optionUnderlier: economicTerms -> payout -> optionPayout only-element -> optionUnderlier
+    alias settlementUnderlier:
+        economicTerms -> payout -> settlementPayout only-element -> underlier
 
     set is_product:
-        economicTerms -> payout -> forwardPayout -> underlier -> index -> ForeignExchangeRate exists
-            or economicTerms -> payout -> optionPayout -> underlier -> index -> ForeignExchangeRate exists
-            or economicTerms -> payout -> performancePayout -> observationTerms -> observable -> index -> ForeignExchangeRate exists
-            or optionUnderlier -> index ->> assetClass = AssetClassEnum -> ForeignExchange
-            or optionUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> ForeignExchange
-            or (if optionUnderlier exists
-                then Qualify_AssetClass_ForeignExchange(
-                            optionUnderlier -> contractualProduct -> economicTerms
-                        )
-                else False)
-            or (if optionUnderlier exists
-                then Qualify_AssetClass_ForeignExchange(
-                            optionUnderlier -> security -> economicTerms
-                        )
-                else False)
-            or forwardUnderlier -> index ->> assetClass = AssetClassEnum -> ForeignExchange
-            or forwardUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> ForeignExchange
-            // legacy
-            or economicTerms -> payout -> forwardPayout -> underlier -> foreignExchange exists
-            or economicTerms -> payout -> optionPayout -> underlier -> foreignExchange exists
+        (economicTerms -> payout -> settlementPayout only exists     
+                and economicTerms -> payout -> settlementPayout -> underlier -> Observable -> Asset -> Cash only exists)
+            or economicTerms -> payout -> performancePayout -> underlier -> Observable -> Index -> ForeignExchangeRateIndex exists
+            or (economicTerms -> payout -> optionPayout only exists
+                and (optionUnderlier -> Observable -> Asset -> Cash only exists
+                    or if optionUnderlier -> Product exists
+                        then Qualify_AssetClass_ForeignExchange(optionUnderlier -> Product ->> economicTerms) = True
+                        else False
+                ))
+            or UnderlierQualification(settlementUnderlier, empty, AssetClassEnum -> ForeignExchange)
```

Comment:
Consider moving this statement under the " settlement payout only" case.

Reply:
done